lib/noise.nit

```diff
@@ -0,0 +1,333 @@
+# It is seeded with the local `seed` and can be further customized by
+# redefining `pseudo_random`.
+# This process do not require any state, so this class only holds the
+# attributes of the generator and does not keep any generated data.
+#
+# ## Usage example
+#
+# ~~~
+# var map = new PerlinNoise
+# map.min = 0.0
+# map.max = 16.0
+# map.period = 20.0
+# map.seed = 0
+#
+# var max = 0.0
+# var min = 100.0
+# for y in 30.times do
+#     for x in 70.times do
+#         # Get a value at x, y
+#         var val = map[x.to_f, y.to_f]
+#         printn val.to_i.to_hex
+#
+#         max = max.max(val)
+#         min = min.min(val)
+#     end
+#     print ""
+# end
+# assert max <= map.max
+# assert min >= map.min
+# ~~~
+#
+# ## Result at seed == 0
+#
+# ~~~raw
+# 7778887888899999888777666666666666666667777877776666666677777777788888
+# 7788877788899999998877776666666666666667777777766666666667777777778888
+# 77877777888999aa999888777777776666666667777777666666666666667777778888
+# 7777777888899aaaa99988877777776666666667777777666666666666667777778888
+# 7777777888899aaaa99998877777776666666667777777666666666666666777778888
+# 7777778888999aaaa99999887777777666666666777667666666666656667777788888
+# 7777778888899aaaa99999888877776666666666666666666666666555666777888888
+# 7777778888999aaaa99999888887777666666666666666666666666555666778888888
+# 6677778888999aaaa99999888887777666666677766666666666666555667778888888
+# 6677788889999aaa999998888877777766666677766667666666666656667788888888
+# 777788889999999a999998888777777766666677777777766666666666667788888888
+# 6677888899999999999988888777667776677777777777766666666666666788888888
+# 6677888999999999999988888776667777777777777777766666665566666788888888
+# 6777788999999999998888887766667777777777777777666666665566666788888889
+# 6777788999999999998888877766677777777777777777667666655566666788988999
+# 7777888999988999988888877766677777777777877777777665555666667788999999
+# 6778888899888889988888877766666777777777777776666665555666667888999999
+# 7788888899888888888888877666666777777777777766666665556666677888898999
+# 7788888899888888888888776666666777777777777666666666566666677888899999
+# 7788889998888888888877766666666777777777777666666666666666777888899999
+# 7888999988888888888777666666666777777777776666666666666667777788899988
+# 7888999988888887777776666666666777777777776666766666666677788888899988
+# 7788999998888877777776666666666777777777776667776666667778888888899988
+# 7788999988888777777776666666666777776677766777776677777788888889999988
+# 6788999988888777777766666666666677676677667777766677777888999999999998
+# 7788888888888777777776666666666666676677777776666777778889999999999998
+# 6778888888888777777766666566666666667777666776667777888999999999999999
+# 6778888888888777777766666566666666667776666776667778889999999999999999
+# 6778888888777777777666666566666666667776666777677788899999999999888999
+# 6777787777777777776666666666666666677777677777777888999999999998888899
```

Comment:
I was looking for pretty ones at first... but I settled on those at seed = 0 for simplicity. :disappointed: